### PR TITLE
MONGOID-5428 Make feature flag connected to Mongo.broken_view_options

### PR DESCRIPTION
**Pull request by Neilshweky**
This PR won't work until 2.18 is released. Driver master works! That's a good sign.

### CODE REVIEW

File: lib/mongoid/config/options.rb [hunk 1]
```diff
@@ -36,8 +36,16 @@ def option(name, options = {})
             end
           end
 
-          define_method("#{name}=") do |value|
-            settings[name] = value
+          if name.to_sym == :broken_view_options
```
Comment (p-mongo) [inline lib/mongoid/config/options.rb:39]: I would do something like `name = name.to_sym` one time in this method right in the beginning.

Comment (p-mongo) [reply lib/mongoid/config/options.rb:39]: Also, the docstring says name can only be a symbol, so either this conversion is unnecessary or the docstring should be amended to specify string as allowed input.

File: lib/mongoid/config/options.rb [hunk 2]
```diff
@@ -36,8 +36,16 @@ def option(name, options = {})
...
+            Mongo.send("#{name}=", options[:default])
+            define_method("#{name}=") do |value|
+              Mongo.send("#{name}=", value)
+              settings[name] = value
```
Comment (p-mongo) [inline lib/mongoid/config/options.rb:43]: If possible I wouldn't have a setting in Mongoid at all, this way the two can't get out of sync.

File: spec/mongoid/config_spec.rb [hunk 3]
```diff
@@ -233,6 +233,12 @@
       end
     end
 
+    around do |ex|
+      temp = Mongoid.send(option)
+      ex.run
+      Mongoid.send("#{option}=", temp)
```
Comment (p-mongo) [inline spec/mongoid/config_spec.rb:239]: I would write a macro similar to the one we have for Mongoid options.

Comment (Neilshweky) [reply spec/mongoid/config_spec.rb:239]: That would be almost the same as config_override. The difference in this case is that the Mongoid options are being modified in other ways within the tests (using loaded configs), and I just want to put it back to what it was before the test.

File: lib/mongoid/config/options.rb (hunk 2 above)
Comment (Neilshweky) [reply lib/mongoid/config/options.rb:43]: Well, I want the default to be true in Mongoid and false in mongo-ruby-driver.... how else would you do that?

File: lib/mongoid/config/options.rb (hunk 1 above)
Comment (Neilshweky) [reply lib/mongoid/config/options.rb:39]: Probably the former.

Comment (Neilshweky) [reply lib/mongoid/config/options.rb:39]: done

**Comment by Neilshweky** [comment]
@p-mongo I made some changes: 

I have fully delegated the `Mongoid.broken_view_options` flag to the `Mongo.broken_view_options flag`, so that when one changes so does the other. The only problem is that when modifying the `Mongo.broken_view_options`, the `Mongoid::Config.settings` does not include the new value (and vice versa). Which could be viewed as problematic. I'm not really sure how to navigate this correctly.

**Review by p-mongo — APPROVED** [review]

**Comment by p-mongo** [comment]
> The only problem is that when modifying the Mongo.broken_view_options, the Mongoid::Config.settings does not include the new value (and vice versa).

In the `settings` method augment the internal hash with the value from driver.

**Comment by p-mongo** [comment]
Actually the settings are mutable so that won't work quite so easily.

**Comment by Neilshweky** [comment]
> Actually the settings are mutable so that won't work quite so easily.

right

**Comment by Neilshweky** [comment]
I thought about excluding it from the settings entirely, but that caused validation errors... 

I have documented this oddity if you want to keep current behavior (updating the Mongo flag doesn't update Mongoid settings)

**Pull request closed without merging**